staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmapleaselock.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcelock
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ConfigMapLeaseLock is used for migration between the configmap and the lease
+type ConfigMapLeaseLock struct {
+	ConfigMapLock *ConfigMapLock
+	LeaseLock     *LeaseLock
+}
+
+// Get returns the older election record of the configmap and the lease
+func (cmll *ConfigMapLeaseLock) Get() (*LeaderElectionRecord, error) {
+	configMapLock, err := cmll.ConfigMapLock.Get()
+	if err != nil {
+		return nil, err
+	}
+	leaseLock, err := cmll.LeaseLock.Get()
+	if err != nil {
+		return nil, err
+	}
+	// check if the older resource lock is released
+	if leaseLock.RenewTime.Before(&configMapLock.RenewTime) {
+		return leaseLock, nil
```

Comment:
this doesn't check that the HolderIdentity agrees between the two... don't that mask scenarios where both objects exist but have conflicting identities recorded?

Reply:
The value of HolderIdentity is informational only. The client doesn't actually act on it.

Reply:
If we return an error when HolderIdentity is different between the two, the function tryAcquireOrRenew will return at first step.
https://github.com/kubernetes/client-go/blob/4f902818859a531e018f805b2d3597f14bf2c494/tools/leaderelection/leaderelection.go#L331
As a result, no client can acquire the leader lease successfully.
Correct me if I am wrong.